Formula/g/gnu-cobol.rb

```diff
@@ -20,40 +24,97 @@ class GnuCobol < Formula
     sha256 x86_64_linux:   "4bf957f586d155f1034ac90ac24edff1253e9de935336680689d85f0bbdcf45c"
   end
 
+  head do
+    # there's no offical git mirror, so we use the current upstream SVN repo
+    url "https://svn.code.sf.net/p/gnucobol/code/trunk"
+
+    # tools for bootstrapping
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "bison" => :build
+    depends_on "flex" => :build
+    depends_on "gettext" => :build
+    depends_on "help2man" => :build
+    depends_on "libtool" => :build
+    depends_on "texinfo" => :build
+
+    # head features multiple indexed handlers
+    depends_on "lmdb"
+    depends_on "unixodbc"
+    # TODO: add "visam" and --with-visam, once formula is added
+
+    # head should be tested before installation
+    uses_from_macos "curl"
+    uses_from_macos "perl"
+  end
+
+  depends_on "pkg-config" => :build
   depends_on "berkeley-db"
   depends_on "gmp"
+  depends_on "json-c"
 
-  def install
-    # both environment variables are needed to be set
-    # the cobol compiler takes these variables for calling cc during its run
-    # if the paths to gmp and bdb are not provided, the run of cobc fails
-    gmp = Formula["gmp"]
-    bdb = Formula["berkeley-db"]
-    ENV.append "CPPFLAGS", "-I#{gmp.opt_include} -I#{bdb.opt_include}"
-    ENV.append "LDFLAGS", "-L#{gmp.opt_lib} -L#{bdb.opt_lib}"
+  uses_from_macos "libxml2"
+  uses_from_macos "ncurses"
 
+  def install
     # Avoid shim references in binaries on Linux.
     ENV["LD"] = "ld" unless OS.mac?
 
-    system "./configure", "--disable-debug",
-                          "--disable-dependency-tracking",
+    indexed_args = ["--with-db"]
+
+    if build.head?
+      indexed_args << "--with-lmdb"
+      indexed_args << "--with-odbc"
+      indexed_args << "--with-indexed=db"
+
+      # go with whatever gettext we have for head
+      system "sed", "-i", "s/AM_GNU_GETTEXT_VERSION/AM_GNU_GETTEXT_REQUIRE_VERSION/", "configure.ac"
```

Comment:
This `sed -i` syntax is not compatible with the BSD `sed` on macOS.

```suggestion
      inreplace "configure.ac", "AM_GNU_GETTEXT_VERSION", "AM_GNU_GETTEXT_REQUIRE_VERSION"
```

Reply:
cool, that also looks better and is shorter, I'll commit that
(maybe something for the automated test to check?)